lib/vanagon/component.rb

```diff
@@ -383,6 +383,23 @@ def get_patches(patch_root)
       end
     end
 
+    # Force version determination for components
+    #
+    # If the component doesn't already have a version set (which normally happens for git sources),
+    # the source will be fetched into a temporary directory to attemt to figure out the version if the
+    # source type supports :version. This directory will be cleaned once the get_sources method returns
+    #
+    # @raise Vanagon::Error raises a vanagon error if we're unable to determine the version
+    def force_version
+      if @version.nil?
+        Dir.mktmpdir do |dir|
+          get_source(dir)
+        end
+      end
+      raise Vanagon::Error, "Unable to determine source version!" if @version.nil?
```

Comment:
Would it be helpful to specify for which component vanagon cannot determine the source version for or would that be obvious in some other way?